lib/matplotlib/widgets.py

```diff
@@ -4227,24 +4230,20 @@ def onrelease(self, event):
         if self.ignore(event):
             return
         if self.verts is not None:
-            self.verts.append((event.xdata, event.ydata))
+            self.verts.append(self._get_data_coords(event))
             if len(self.verts) > 2:
                 self.callback(self.verts)
             self.line.remove()
         self.verts = None
         self.disconnect_events()
 
     def onmove(self, event):
-        if self.ignore(event):
-            return
-        if self.verts is None:
-            return
-        if event.inaxes != self.ax:
-            return
-        if event.button != 1:
+        if (self.ignore(event)
+                or self.verts is None
+                or not self.ax.contains(event)[0]
+                or event.button != 1):
```

Comment:
The button check is probably (slightly) less expensive than the `contains`:
```suggestion
                or event.button != 1
                or not self.ax.contains(event)[0]):
```